axiom/tests/test_log_events.c

```diff
@@ -142,33 +161,125 @@ static void test_events_sample(void) {
   tlib_pass_if_int_equal("events number saved updated", 2,
                          nr_log_events_number_saved(events));
 
-  nr_random_destroy(&rnd);
+  // verify two highest priority events were retained
+  // convert log event pool to a vector and test
```

Comment:
I'm not sure these tests should exists. nr_minmax_heap_iterate (underlying implementation of nr_log_events_to_vector) documentation says:

> Note : Ordering is not guaranteed; calling code should be prepared to
receive values in any order.

I'm afraid this may make this test unpredictable. I don't know how to make this test independent of the order of elements in the vector.

Reply:
I've changed the test to sort the vector using the same comparator as the heap uses.  This should put the event with the minimum priority in position 0 and the other event at position 1 reliably.

Fix in commit [284f848](https://github.com/newrelic/newrelic-php-agent/pull/515/commits/284f848717cb9c0eef21f25b75899379077f9ceb).